### PR TITLE
FiBo-139: mark invalid fields in adding cashflow form

### PR DESCRIPTION
**Pull request by XSnelliusX**

### CODE REVIEW

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]
Overall it works fine. 

There are 2 changes that I suggest: 

1: please change the method checkValidInput:
```
        fieldsToBeChecked.keySet()
                .forEach(field -> field.setErrorEnabled(false));
```
In that way. The field layout height will reset after each check. Otherwise the gap between each row will stay, even when the error message is cleared.

2nd: 

I think now is the right time to delete the attachment button behind the Categorie dropdown. It is jumping around when there is an error in the categorie layout, because the height of the layout increases.

**Comment by XSnelliusX** [comment]
I implemented the requested changes, and added an test as well, an other review would be nice ;)

**Review by jens1o — CHANGES_REQUESTED** [review]
could you please reformat all files? :)

File: android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java
Comment (Jonas-Karl) [inline android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java:240]:
```suggestion
        dateText.setFocusable(View.NOT_FOCUSABLE);
        dateText.setOnClickListener(showDatePicker());
        binding.dateLayout.setEndIconOnClickListener(showDatePicker());
    }

    private View.OnClickListener showDatePicker() {
        return (view) -> datePicker.show(requireActivity().getSupportFragmentManager(), "datePick");
    }
    }
```

Can be shortened to this

File: android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java
Comment (Jonas-Karl) [inline android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragment.java:167]: This line is not needed anymore. You can check it in line 173 with `if (checkForRequiredData())` instead of `if (isRequiredDataPresent)`

File: android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java
Comment (Jonas-Karl) [inline android-app/app/src/androidTest/java/de/dhbw/ka/se/fibo/ui/adding/AddingFragmentTest.java:58]: Here it would be useful to split the test a bit. There are many scenarios testet in one single test. 
e.g. `testInvalidIncomeInputs()`, `testInvalidExpenseInputs()`, `testMultipleInvalidInputs()`.
Obviously, testing for every combination of inputs seems to be a bit much right now, but it would be helpful if there are more smaller tests instead of one big test

**Review by Jonas-Karl — CHANGES_REQUESTED** [review]
Well done with implementing the changes. At this point I think it can be merged if you want to. Anyways, I think there are still some minor code quality improvements possible as I mentioned in the comments of this review.

Feel free to implement them before merging :)

**Review by Jonas-Karl — APPROVED** [review]
I found a small issue that was introcuded in this PR but I created a Ticket for it as it is a minor improvement. I think this one is ready to merge :)

Here is the ticket: https://dhbw-karlsruhe.myjetbrains.com/youtrack/issue/FiBo-340/Update-adding-fragment-error-text-when-switching-between-selected-tabs

**Review by jens1o — APPROVED** [review]
lgtm, thanks!

**Pull request merged**
